internal/util/serdeio.go

```diff
@@ -79,6 +80,11 @@ func newYAMLDecoder(src *bufio.Reader) decoderFunc {
 		for s.Scan() {
 			line := s.Bytes()
 
+			// ignore comments
+			if bytes.HasPrefix(line, yamlComment) {
```

Comment:
Perhaps unlikely, but I think it's valid for the # to be preceded by whitespace.